src/Sarif.Driver/Sdk/Errors.cs

```diff
@@ -3,234 +3,185 @@
-            // {1}
-            context.Logger.Log(context.Rule,
-                RuleUtilities.BuildResult(ResultKind.ConfigurationError, context, null,
-                    nameof(SdkResources.ERR0997_ExceptionAccessingFile),
-                    fileName,
-                    ex.ToString()));
+            // Could not access a file specified on the command-line: "{0}".
+            context.Logger.LogConfigurationNotification(
+                CreateNotification(
+                    Notification_ExceptionAccessingFile,
+                    NotificationLevel.Error,
+                    exception,
+                    fileName));
 
             context.RuntimeErrors |= RuntimeConditions.ExceptionAccessingFile;
         }
 
-
         public static void LogMissingRuleConfiguration(IAnalysisContext context, string reasonForNotAnalyzing)
         {            
-            string ruleName = context.Rule.Name;
-            context.Rule = Errors.InvalidConfiguration;
-
             Assembly assembly = Assembly.GetEntryAssembly();
             assembly = assembly ?? Assembly.GetExecutingAssembly();
             string exeName = Path.GetFileName(assembly.Location);
 
-            // Check '{1}' was disabled while analyzing '{0}' because the analysis
-            // was not configured with required policy ({1}). To resolve this,
-            // configure and provide a policy file on the {2} command-line using
+            // Check '{0}' was disabled while analyzing '{1}' because the analysis
+            // was not configured with required policy ({2}). To resolve this,
+            // configure and provide a policy file on the {3} command-line using
             // the --policy argument (recommended), or pass '--config default'
-            // to invoke built-in settings. Invoke the {2} 'exportConfig' command
+            // to invoke built-in settings. Invoke the {3} 'exportConfig' command
             // to produce an initial configuration file that can be edited, if
             // necessary, and passed back into the tool.
-            context.Logger.Log(context.Rule,
-                RuleUtilities.BuildResult(ResultKind.ConfigurationError, context, null,
-                    nameof(SdkResources.ERR0997_MissingRuleConfiguration),
-                    ruleName,
-                    reasonForNotAnalyzing,
-                    exeName));
+            context.Logger.LogConfigurationNotification(
+                CreateNotification(
+                    Notification_MissingRuleConfiguration,
+                    NotificationLevel.Error,
+                    null,
+                    context.Rule.Name,
+                    string.Empty,           // BUG: There were fewer arguments specified than required by the format string
+                    reasonForNotAnalyzing,  // ... and it doesn't look like this fits with the message for {2}
+                    exeName));              // ... but this is pretty clearly {3}.
```

Comment:
Please take a look at this call (and my comments on it). I think it was buggy before, and I don't know what the message and its parameters are really supposed to be.